tests/benchmark/app/App.ts

```diff
@@ -0,0 +1,124 @@
+
+import { DNode } from '../../../src/interfaces';
+import { WidgetBase } from '../../../src/WidgetBase';
+import { v, w } from '../../../src/d';
+
+import { Row } from './Row';
+import { Buttons, ButtonConfig } from './Buttons';
+import { Store } from  './Store';
+
+let startTime: number;
+let lastMeasure: string | null;
+
+function startMeasure(name: string) {
+	startTime = performance.now();
+	lastMeasure = name;
+}
+
+function stopMeasure() {
+	const last = lastMeasure;
+	if (lastMeasure) {
+		setTimeout(function () {
+			lastMeasure = null;
+			const stop = performance.now();
+			console.log(`${last} took ${(stop - startTime)}`);
+		}, 0);
+	}
+}
+
+export class App extends WidgetBase {
+	private _store: Store = new Store();
+
+	private _printDuration() {
+		stopMeasure();
+	}
+
+	public onElementUpdated() {
+		this._printDuration();
+	}
+
+	public onElementCreated() {
+		this._printDuration();
+	}
+
+	private _run = () => {
+		startMeasure('run');
+		this._store.run();
+		this.invalidate();
+	}
+
+	private _add = () => {
+		startMeasure('add');
+		this._store.add();
+		this.invalidate();
+	}
+
+	private _update = () => {
+		startMeasure('update');
+		this._store.update();
+		this.invalidate();
+	}
+
+	private _select = (id: number) => {
+		startMeasure('select');
+		this._store.select(id);
+		this.invalidate();
+	}
+
+	private _delete = (id: number) => {
+		startMeasure('delete');
+		this._store.delete(id);
+		this.invalidate();
+	}
+
+	private _runLots = () => {
+		startMeasure('runLots');
+		this._store.runLots();
+		this.invalidate();
+	}
+
+	private _clear = () => {
+		startMeasure('clear');
+		this._store.clear();
+		this.invalidate();
+	}
+
+	private _swapRows = () => {
+		startMeasure('swapRows');
+		this._store.swapRows();
+		this.invalidate();
+	}
+
+	private _buttonConfigs: ButtonConfig[] = [
+		{ id: 'run', label: 'Create 1,000 rows', onClick: this._run },
+		{ id: 'runlots', label: 'Create 10,000 rows', onClick: this._runLots },
+		{ id: 'add', label: 'Append 1,000 rows', onClick: this._add },
+		{ id: 'update', label: 'Update every 10th row', onClick: this._update },
+		{ id: 'clear', label: 'Clear', onClick: this._clear },
+		{ id: 'swaprows', label: 'Swap Rows', onClick: this._swapRows }
+	];
+
+	protected render (): DNode {
+		const { _select, _delete, _store } = this;
+		const rows = _store.data.map(({ id, label }, index) => {
+			return w(Row, {
+				id,
+				key: index,
```

Comment:
Let’s make this `id` so it’s a true keyed impl as per the benchmarks definition

Reply:
Done